net/shadowsocks-libev/files/ss-rules

```diff
@@ -120,6 +120,9 @@ tp_rule() {
 	$ipt_m -A SS_SPEC_TPROXY -p udp -m set ! --match-set ss_spec_wan_ac dst \
 		-j TPROXY --on-port $LOCAL_PORT --tproxy-mark 0x01/0x01
 	$ipt_m -A PREROUTING -p udp $EXT_ARGS \
+		-m socket \
+		-m comment --comment "_SS_SPEC_RULE_" -j MARK --set-mark 1
+	$ipt_m -A PREROUTING -p udp $EXT_ARGS \
```

Comment:
I have also found and read these documents before.  But they cannot give me answers to some questions/doubts I have then.

I think we now both agree that the `--tproxy-mark` will mark skb the same way as with `-j MARK` action.  This is how the iptables-extentions doc tells me and how it's written in the kernel xt_TPROXY.c

I was also very curious about the the `-m socket -j DIVERT` rule in the `Documentation/networking/tproxy.txt`.  It turns out that "The only required steps are the routing and the TPROXY iptables-rule, the DIVERT-rule is an optimisation to prevent unnecessary processing of packets in the TPROXY target" (https://github.com/kristrev/tproxy-example/blob/master/README.md)